src/EditorFeatures/CSharpTest/EditAndContinue/StatementEditingTests.cs

```diff
@@ -8344,6 +8344,10 @@ static async Task<int> F()
         var a = F(1, await F(1)), b = F(1, await G(1));
         b = F(1, await F(1));
         b += await F(1);
+        var r = (x, y) switch {
+            (Task<object> obj, 3) => await obj,
+            _ => 0
+        };
```

Comment:
What is this case trying to test? The rest of the cases cover spilling. If that's what you're trying to test than
`F(x switch { 1 => await F(1), _ => 0 });` -> `F(x switch { 1 => await F(2), _ => 0 });`
would be simpler way to do so.

Reply:
It would be useful to add a comment that this test (and the previous one) covers `ReportStateMachineSuspensionPointRudeEdits`

---
In reply to: [301331512](https://github.com/dotnet/roslyn/pull/37015#discussion_r301331512) [](ancestors = 301331512)

Reply:
OK. If you think it is not necessary, removing it.
Added comments

---
In reply to: [301336578](https://github.com/dotnet/roslyn/pull/37015#discussion_r301336578) [](ancestors = 301336578,301331512)